docs/spikes/using-image-data/ai-vision.ipynb

```diff
@@ -0,0 +1,533 @@
+    "    - A `text-embeddings-ada-002` model deployment\n",
+    "    - A `gpt-4-vision` model deployment\n",
+    "- Azure Computer Vision\n",
+    "\n",
+    "You can reuse some of the components deployed as part of the Chat With Your Data project. However, at the time of \n",
+    "writing you will also additionally need to deploy Azure Computer Vision and a `gpt-4-vision ` model.\n",
+    "\n",
+    "Once provisioned, you need then to change the configuration below to match your setup.\n",
+    "\n",
+    "## Data\n",
+    "\n",
+    "This spike only uploads one image, which is a flow chart detailing when to use various Azure services. This image was \n",
+    "sourced from https://learn.microsoft.com/en-us/azure/architecture/guide/technology-choices/compute-decision-tree.\n",
+    "\n",
+    "Feel free to use a different image and question.\n",
+    "\n",
+    "## How it works\n",
+    "\n",
+    "### Ingestion\n",
+    "\n",
+    "To ingest an image into Azure AI Search, the following steps are performed:\n",
+    "1. Create search index\n",
+    "2. Upload image to blob storage\n",
+    "3. Generate embeddings using computer vision `vectorizeImage` API\n",
+    "4. Generate a caption of the image using `gpt-4-vision`\n",
+    "5. Generate embeddings of the caption using `text-embeddings-002-ada`\n",
+    "6. Store data in search index\n",
+    "\n",
+    "\n",
+    "### Question\n",
+    "\n",
+    "To ask a question using this data, the following steps are performed:\n",
+    "1. Generate embeddings for the question using computer vision `vectorizeText` API\n",
+    "2. Generate embeddings for the question using `text-embeddings-002-ada`\n",
+    "3. Search index using both embeddings\n",
+    "4. Generate blob sas url from returned search results\n",
+    "5. Pass question, along with blob sas url to `gpt-4-vision` chat completions end point\n",
+    "\n",
+    "\n",
+    "## Why do we need two different embedding models?\n",
+    "\n",
+    "It is not required to use two different embedding models, however, using both Azure Computer Vision to embed the image\n",
+    "and `gpt-4-vision` to generate a description that is then embedded by `text-embeddings-002-ada` provides richer data and\n",
+    "provides better search results. This is useful in particular for diagrams and flow charts which show relationships and\n",
+    "decision points.\n",
+    "\n",
+    "When embedding the question, we need to embed it with both embeddings models to allow us to search over the previously\n",
+    "generated vectors. This is due to the embeddings models generating differently sized and formatted vectors."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "e0da0335-4787-499a-abe9-bd99971bb98c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Common imports\n",
+    "import re\n",
+    "import datetime\n",
+    "import base64\n",
```

Comment:
nit: you dont use base64 - happy for you to ignore so you dont have to get reapprovals! :-)